src/rbac.rs

```diff
@@ -67,13 +69,9 @@ impl RoleManager for DefaultRoleManager {
         } else if domain.len() > 1 {
             panic!("error domain length");
         }
-        let mut role1 = self.create_role(name1.as_str());
-        let role2 = self.create_role(name2.as_str());
-        role1.add_role(role2);
-        // role1 is updated and should be updated into all_roles
-        if let Some(old_role) = self.all_roles.get_mut(&role1.name) {
-            *old_role = role1.clone();
-        }
+        let role1 = self.create_role(&name1);
+        let role2 = self.create_role(&name2);
+        role1.borrow_mut().add_role(role2.clone());
```

Comment:
Everything looks great if I was going to suggest something and this is a nitpick, the convention is to use Rc::clone(&role2) only because this distinguishes it from normal clone, but it doesn't really matter as long as you know whats going on.

Reply:
You are right, it's better to use `Rc::clone`, fixed in the new commit